service/history/task/transfer_active_task_executor.go

```diff
@@ -1219,7 +1229,12 @@ func (t *transferActiveTaskExecutor) generateCrossClusterTask(
 		wfContext,
 		false,
 		func(ctx context.Context, mutableState execution.MutableState) error {
-			if !mutableState.IsWorkflowExecutionRunning() {
+			if task.TaskType == persistence.TransferTaskTypeCloseExecution {
+				if !mutableState.IsWorkflowCompleted() {
```

Comment:
Do we need this new method? Is `!mutableState.IsWorkflowCompleted` the same as `mutableState.IsWorkflowExecutionRunning`?

Reply:
Not exactly same, do we want to add the new x-cluster task for the following states? 

```
persistence.WorkflowStateZombie:
persistence.WorkflowStateCorrupted:
```

Reply:
If it's not expected, why should we include those cases here? It's probably better to only have the valid case. Having other states can also confuse other people reading here. We can talk today to see if you have other concerns.